### PR TITLE
Gracefully handle missing properties

### PR DESCRIPTION
**Pull request by ploober**
This PR has several goals:
- Gracefully handle missing properties for both Imgur and Discord
- Reduce code duplication within the meme commands
- Add a new rest endpoint for executing commands (unsecured currently)

### CODE REVIEW

File: src/main/java/com/gorlah/kappabot/integration/discord/DiscordBot.java [hunk 1]
```diff
@@ -19,74 +20,75 @@
...
-                
+
                 response.append("> ").append(titles.get(i));
             }
         } else {
             if (message.length() < commandPrefix.length()
-                    || !commandPrefix.equalsIgnoreCase(message.substring(0, commandPrefix.length()))) {
+                    || !commandPrefix.equalsIgnoreCase(message.substring(
+                    0,
+                    commandPrefix.length()))) {
```
Comment (erichaagdev) [inline src/main/java/com/gorlah/kappabot/integration/discord/DiscordBot.java]: Was this code styling intended?

File: src/main/java/com/gorlah/kappabot/integration/discord/DiscordBot.java [hunk 2]
```diff
@@ -19,74 +20,75 @@
...
                 return;
             }
-            
-            Command command = new Command(commandPrefix, message, event.getAuthor().getName());
-    
+
+            Command command =
+                    new Command(commandPrefix, message, event.getAuthor().getName());
```
Comment (erichaagdev) [inline src/main/java/com/gorlah/kappabot/integration/discord/DiscordBot.java]: I don't think this line is long enough to warrant a break.  Was this intended?  Is this that the project code style can be updated for?

Comment (ploober) [reply src/main/java/com/gorlah/kappabot/integration/discord/DiscordBot.java]: I don't know why this one got wrapped, I didn't specifically do it.

File: src/main/java/com/gorlah/kappabot/integration/discord/DiscordBot.java (hunk 1 above)
Comment (ploober) [reply src/main/java/com/gorlah/kappabot/integration/discord/DiscordBot.java]: That's just how the code style rules decided to chop it down, I will reformat it in a more pleasing way.

Comment (erichaagdev) [reply src/main/java/com/gorlah/kappabot/integration/discord/DiscordBot.java]: Okay so I guess the code style needs to be updated then.  I can look into that.  Thanks for fixing it.

File: src/main/java/com/gorlah/kappabot/integration/discord/DiscordBot.java (hunk 2 above)
Comment (erichaagdev) [reply src/main/java/com/gorlah/kappabot/integration/discord/DiscordBot.java]: Weird.  I'll double check the code style.  Thanks!

**Pull request merged**